src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,296 @@
+     * @param state .
+     * @param property .
+     * @param value .
+     * @return .
+     * @param <T> .
+     */
+    private static <T extends Comparable<T>> IBlockState setPropertyValue(IBlockState state, IProperty<T> property, String value) {
+        Optional<T> parsed = property.parseValue(value).toJavaUtil();
+        if (parsed.isPresent()) {
+            return state.withProperty(property, parsed.get());
+        } else {
+            throw new IllegalArgumentException("Invalid value for property " + property);
+        }
+    }
+
+    /**
+     * @param amountOfBlockTypes amount of block types in the schematic.
+     * @return amount of bits used to encode a block.
+     */
+    private static int getBitsPerBlock(int amountOfBlockTypes) {
+        return  (int) Math.floor((Math.log(amountOfBlockTypes)) / Math.log(2))+1;
+    }
+
+    /**
+     * @return the volume of the subregion.
+     */
+    private static long getVolume(NBTTagCompound nbt, String subReg) {
+        return Math.abs(
+                nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(size).getInteger("x") *
+                        nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(size).getInteger("y") *
+                        nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(size).getInteger("z"));
+    }
+
+    /**
+     * @return array of Long values.
+     */
+    private static long[] getBlockStates(NBTTagCompound nbt, String subReg) {
+        return ((INBTTagLongArray) nbt.getCompoundTag(reg).getCompoundTag(subReg).getTag(blSt)).getLongArray();
+    }
+
+    /**
+     * @param blockList list with the different block types used in the schematic
+     * @param bitArray bit array that holds the placement pattern
+     */
+    private void writeSubregionIntoSchematic(NBTTagCompound nbt, String subReg, IBlockState[] blockList, LitematicaBitArray bitArray) {
+        int posX = getMinimumCord(nbt, subReg,"x");
+        int posY = getMinimumCord(nbt, subReg,"y");
+        int posZ = getMinimumCord(nbt, subReg,"z");
+        int index = 0;
+        for (int y = 0; y < this.y; y++) {
+            for (int z = 0; z < this.z; z++) {
+                for (int x = 0; x < this.x; x++) {
+                    if (inSubregion(nbt, subReg, x, y, z)) {
+                        this.states[x-(minX- posX)][z-(minZ- posZ)][y-(minY- posY)] = blockList[bitArray.getAt(index)];
+                        index++;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param x cord of the schematic.
+     * @param y cord of the schematic.
+     * @param z cord of the schematic.
+     * @return if the current block is inside the subregion.
+     */
+    private static boolean inSubregion(NBTTagCompound nbt, String subReg, int x, int y, int z) {
+        return
+                x < Math.abs(nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(size).getInteger("x")) &&
```

Comment:
how does this work if size can be negative? what if x is negative too?

Reply:
size can be negative but we take the absulute value to compare it, makeing it allways positive.